src/transformers/models/imagebind/modeling_imagebind.py

```diff
@@ -0,0 +1,2101 @@
+                position_ids=position_ids,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+                return_dict=return_dict,
+            )
+            text_embeds = text_outputs[1]
+            text_embeds = self.text_projection(text_embeds)
+            text_embeds = self.text_postprocessor(text_embeds)
+
+            logits_per_text = torch.matmul(text_embeds, image_embeds.t())
+            logits_per_image = logits_per_text.t()
+
+        # running the audio model if input_features is provided
+        audio_embeds = None
+        logits_per_audio = None
+        audio_outputs = None
+        if input_features is not None:
+            audio_batch_size = input_features.shape[0]
+            audio_outputs = self.audio_model(
+                input_features,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+                return_dict=return_dict,
+            )
+            audio_embeds = audio_outputs[1]
+            audio_embeds = self.audio_projection(audio_embeds)
+            audio_embeds = self.audio_postprocessor(audio_embeds)
+
+            if input_features.ndim >= 5:
+                num_clips = input_features.shape[1]
+                audio_embeds = audio_embeds.reshape(audio_batch_size, num_clips, -1)
+                audio_embeds = audio_embeds.mean(dim=1)
+
+            logits_per_audio = torch.matmul(audio_embeds, image_embeds.t())
+            logits_per_image = logits_per_audio.t()
+
+        loss = None
+        if return_loss:
+            loss = imagebind_loss(logits_per_text) if logits_per_text is not None else imagebind_loss(logits_per_audio)
+
+        if not return_dict:
+            output = (
+                logits_per_image,
+                logits_per_text,
+                logits_per_audio,
+                image_embeds,
+                text_embeds,
+                audio_embeds,
+                vision_outputs,
+                text_outputs,
+                audio_outputs,
+            )
+            output = tuple([out for out in output if out is not None])
+            return ((loss,) + output) if loss is not None else output
+
+        return ImageBindOutput(
+            loss=loss,
+            logits_per_image=logits_per_image,
+            logits_per_text=logits_per_text,
+            logits_per_audio=logits_per_audio,
+            image_embeds=image_embeds,
+            text_embeds=text_embeds,
+            audio_embeds=audio_embeds,
+            vision_model_output=vision_outputs,
+            text_model_output=text_outputs,
+            audio_model_output=audio_outputs,
+        )
+
+
+@add_start_docstrings(
+    """
+    ImageBind Text Model with a projection layer on top (a linear layer on top of the pooled output).
+    """,
+    IMAGEBIND_START_DOCSTRING,
+)
+class ImageBindTextModelWithProjection(ImageBindPreTrainedModel):
```

Comment:
On the "with projection" classes: are they using the same base model `imagebind-huge` for instance? in that case, it might save some duplicated code to have an init or config key "projection_layer=True` in the main classes

Reply:
Indeed, we could have a `with_projection_layer` in the `ImageBindXxxModel`s, but that would go against `CLIP` and `CLAP` implementations

Reply:
Fair enough, we can keep as it is!